xclim/ensembles/_partitioning.py

```diff
@@ -0,0 +1,194 @@
+Components of an Incomplete Ensemble of Climate Projections Using Data Augmentation, Journal of Climate, 32(8),
+2423-2440, https://doi.org/10.1175/JCLI-D-18-0606.1
+
+Related bixtex entries:
+ - yip_2011
+ - northrop_2014
+ - goldenson_2018
+ - lehner_2020
+ - evin_2019
+"""
+
+# Default dimension names
+_dims = dict(model="model", scenario="scenario", member="member")
+
+
+def hawkins_sutton(
+    da: xr.DataArray,
+    sm: xr.DataArray = None,
+    weights: xr.DataArray = None,
+    baseline: tuple = ("1971", "2000"),
+    kind: str = "+",
+    dimensions: dict = None,
+):
+    """Return the mean and partitioned variance of an ensemble based on method from Hawkins & Sutton.
+
+    Algorithm based on :cite:t:`hawkins_2009`. Input data should meet the following requirements:
+      - annual frequency;
+      - covers the baseline and future period;
+      - defined over time, scenario and model dimensions;
+      - the same models should be available for each scenario.
+
+    Parameters
+    ----------
+    da: xr.DataArray
+      Time series over dimensions of 'time', 'scenario' and 'model'. If other dimension names are used, provide
+      the mapping with the `dimensions` argument.
+    sm: xr.DataArray
+      Smoothed time series over time. By default, the method uses a fourth order polynomial. Use this argument to
+      use other smoothing strategies, such as another polynomial order, or a LOESS curve.
+    weights: xr.DataArray
+      Weights to be applied to individual models.
+    baseline: [str, str]
+      Start and end year of the reference period.
+    kind: {'+', '*'}
+      Whether the mean over the reference period should be substracted (+) or divided by (*).
+    dimensions: dict
+      Mapping from original dimension names to standard dimension names: scenario, model, member.
+
+    Returns
+    -------
+    xr.DataArray, xr.DataArray
+      The mean and the components of variance of the ensemble. These components are coordinates along the
+      `uncertainty` dimension: `variability`, `model`, `scenario`, and `total`.
+
+    References
+    ----------
+    :cite:cts:`hawkins_2009,hawkins_2011`
+    """
+    if xr.infer_freq(da.time)[0] not in ["A", "Y"]:
+        raise ValueError("This algorithm expects annual time series.")
+
+    if dimensions is None:
+        dimensions = {}
+
+    da = da.rename(reverse_dict(dimensions))
+
+    if "member" in da.dims and len(da.member) > 1:
+        raise ValueError("There should only be one member per model.")
+
+    # Confirm the same models have data for all scenarios
+    check = da.notnull().any("time").all("scenario")
+    if not check.all():
+        raise ValueError(f"Some models are missing data for some scenarios: \n {check}")
+
+    if weights is None:
+        weights = xr.ones_like(da.model, float)
+
+    # Perform analysis 1950 onward
+    da = da.sel(time=slice("1950", None))
```

Comment:
Seems strange to me to hardcode this? Or is it a condition from the literature?

Reply:
Yep, it's from the paper. Again, it's the question of how close should these algos stick to the papers they're based on.  No worries adding an argument for this.

Reply:
I think I like this last idea!